src/main/java/seedu/workbook/model/internship/UniqueInternshipList.java

```diff
@@ -104,31 +122,78 @@ public void setInternships(List<Internship> internships) {
             throw new DuplicateInternshipException();
         }
 
-        internalList.setAll(internships);
+        baseInternalList.setAll(internships);
+        refreshUiFacingInternalList();
+    }
+
+    /**
+     * Toggles sorting by date.
+     */
+    public void sortByDate() {
+        // changes active view to the sorted view
+        currentlySortedByDate = true;
+        refreshUiFacingInternalList();
+    }
+
+    /**
+     * Resets order of list, aka back to order in which Internships were added.
+     */
+    public void resetOrder() {
+        currentlySortedByDate = false;
+        refreshUiFacingInternalList();
+    }
+
+    /**
+     * (Re)Populates the base uiFacingInternalList with the correct internalList
+     * based on sort flag.
+     * Called whenever underlying `baseInternalList` has been modified, and allows
+     * user to see changes even if in sorted view.
+     * Recall that UI components are observing the unmodifiable wrapper on
+     * this `uiFacingInternalList`.
+     *
+     * @see #asUiFacingUnmodifiableObservableList()
+     */
+    private void refreshUiFacingInternalList() {
+        if (currentlySortedByDate) {
+            uiFacingInternalList.setAll(internalListSortedByDate);
+        } else {
+            uiFacingInternalList.setAll(baseInternalList);
+        }
```

Comment:
Good use of JavaDocs to explain the implementation. I find this method well implemented too.